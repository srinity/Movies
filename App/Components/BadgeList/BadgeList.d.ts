import { Component } from 'react'
import { TextProps, ViewStyle, TextStyle, StyleProp } from 'react-native'

export type BadgeItem = {
  id: number
  name: string
}

export interface BadgeListProps extends Omit<TextProps, 'style'> {
  /**
   * The name that will appear in the badge.
   *
   */
  data: BadgeItem[]

  /**
   * optional custom styles for the container.
   *
   */
  containerStyle: StyleProp<ViewStyle>

  /**
   * optional custom styles for the item container.
   *
   */
  itemStyle: StyleProp<ViewStyle>

  /**
   * optional custom styles for the each item text
   *
   */
  itemTextStyle: StyleProp<TextStyle>
}

export default class BadgeList extends Component<BadgeListProps> {}
