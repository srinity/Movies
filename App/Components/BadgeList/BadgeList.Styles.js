import { StyleSheet } from 'react-native'

export default StyleSheet.create({
  containerStyle: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    width: '100%'
  }
})
