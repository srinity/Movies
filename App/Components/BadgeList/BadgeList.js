import React from 'react'
import { View, Text, ViewPropTypes } from 'react-native'
import PropTypes from 'prop-types'
import { map as _map } from 'lodash'

import Badge from '../Badge'

import styles from './BadgeList.Styles'

function BadgeList({
  data,
  containerStyle,
  itemStyle,
  itemTextStyle,
  ...props
}) {
  return (
    <View style={[styles.containerStyle, containerStyle]}>
      {_map(data, item => (
        <Badge
          key={item.id}
          style={itemStyle}
          textStyle={itemTextStyle}
          {...item}
        />
      ))}
    </View>
  )
}

BadgeList.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      id: PropTypes.number
    })
  ),
  containerStyle: ViewPropTypes.style,
  itemStyle: ViewPropTypes.style,
  itemTextStyle: Text.propTypes.style
}

export default BadgeList
