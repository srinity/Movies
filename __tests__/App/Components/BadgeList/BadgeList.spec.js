import React from 'react'
import { Text, View } from 'react-native'
import renderer from 'react-test-renderer'

import BadgeList from '../../../../App/Components/BadgeList/BadgeList'

describe('BadgeList Component', () => {
  const data = [
    {
      name: 'Actions',
      id: 1
    },
    {
      name: 'Animation',
      id: 2
    }
  ]

  it('should render correctly', () => {
    const badgeList = renderer.create(<BadgeList data={data} />)
    expect(badgeList).toMatchSnapshot()
  })

  it('should be able to customize itemTextStyle correctly', () => {
    const textStyle = {
      color: '#e23478'
    }

    const badgeList = renderer.create(
      <BadgeList data={data} itemTextStyle={textStyle} />
    )
    expect(badgeList).toMatchSnapshot()

    expect(badgeList.root.findAllByType(Text)[0].props.style[1]).toBe(textStyle)
  })

  it('should be able to customize itemStyle correctly', () => {
    const style = {
      backgroundColor: '#e23478'
    }

    const badgeList = renderer.create(
      <BadgeList data={data} itemStyle={style} />
    )
    expect(badgeList).toMatchSnapshot()

    expect(badgeList.root.findAllByType(View)[1].props.style[1]).toBe(style)
  })

  it('should be able to customize containerStyle correctly', () => {
    const style = {
      backgroundColor: '#e23478'
    }

    const badgeList = renderer.create(
      <BadgeList data={data} containerStyle={style} />
    )
    expect(badgeList).toMatchSnapshot()

    expect(badgeList.root.findByType(View).props.style[1]).toBe(style)
  })
})
