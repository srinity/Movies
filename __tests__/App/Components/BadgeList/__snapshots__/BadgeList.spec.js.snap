// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeList Component should be able to customize containerStyle correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "width": "100%",
      },
      Object {
        "backgroundColor": "#e23478",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#dddddd",
          "borderRadius": 12.5,
          "marginRight": 5,
          "marginTop": 7,
          "paddingHorizontal": 10,
          "paddingVertical": 2,
        },
        undefined,
      ]
    }
  >
    <Text
      id={1}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 10,
          },
          undefined,
        ]
      }
    >
      Actions
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#dddddd",
          "borderRadius": 12.5,
          "marginRight": 5,
          "marginTop": 7,
          "paddingHorizontal": 10,
          "paddingVertical": 2,
        },
        undefined,
      ]
    }
  >
    <Text
      id={2}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 10,
          },
          undefined,
        ]
      }
    >
      Animation
    </Text>
  </View>
</View>
`;

exports[`BadgeList Component should be able to customize itemStyle correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#dddddd",
          "borderRadius": 12.5,
          "marginRight": 5,
          "marginTop": 7,
          "paddingHorizontal": 10,
          "paddingVertical": 2,
        },
        Object {
          "backgroundColor": "#e23478",
        },
      ]
    }
  >
    <Text
      id={1}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 10,
          },
          undefined,
        ]
      }
    >
      Actions
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#dddddd",
          "borderRadius": 12.5,
          "marginRight": 5,
          "marginTop": 7,
          "paddingHorizontal": 10,
          "paddingVertical": 2,
        },
        Object {
          "backgroundColor": "#e23478",
        },
      ]
    }
  >
    <Text
      id={2}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 10,
          },
          undefined,
        ]
      }
    >
      Animation
    </Text>
  </View>
</View>
`;

exports[`BadgeList Component should be able to customize itemTextStyle correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#dddddd",
          "borderRadius": 12.5,
          "marginRight": 5,
          "marginTop": 7,
          "paddingHorizontal": 10,
          "paddingVertical": 2,
        },
        undefined,
      ]
    }
  >
    <Text
      id={1}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 10,
          },
          Object {
            "color": "#e23478",
          },
        ]
      }
    >
      Actions
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#dddddd",
          "borderRadius": 12.5,
          "marginRight": 5,
          "marginTop": 7,
          "paddingHorizontal": 10,
          "paddingVertical": 2,
        },
        undefined,
      ]
    }
  >
    <Text
      id={2}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 10,
          },
          Object {
            "color": "#e23478",
          },
        ]
      }
    >
      Animation
    </Text>
  </View>
</View>
`;

exports[`BadgeList Component should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#dddddd",
          "borderRadius": 12.5,
          "marginRight": 5,
          "marginTop": 7,
          "paddingHorizontal": 10,
          "paddingVertical": 2,
        },
        undefined,
      ]
    }
  >
    <Text
      id={1}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 10,
          },
          undefined,
        ]
      }
    >
      Actions
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#dddddd",
          "borderRadius": 12.5,
          "marginRight": 5,
          "marginTop": 7,
          "paddingHorizontal": 10,
          "paddingVertical": 2,
        },
        undefined,
      ]
    }
  >
    <Text
      id={2}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 10,
          },
          undefined,
        ]
      }
    >
      Animation
    </Text>
  </View>
</View>
`;
